src/vtquery.cpp

```diff
@@ -163,12 +190,97 @@ struct CompareDistance {
     }
 };
 
+std::vector<vtzero::index_value_pair> get_comparison_tags(vtzero::feature f) {
+    std::vector<vtzero::index_value_pair> v;
```

Comment:
Adding `v.reserve(f.num_properties())` here should, I presume,  improve performance by pre-allocating once and avoiding re-allocation as the vector grows (and resizes).